metaci/release/tasks.py

```diff
@@ -310,6 +320,120 @@ def all_deps_satisfied(
+                        AllPackageVersionId=all_package_version_id,
+                        Dependencies=[],
+                    )
+
+
+def convert_dependency_graph_to_metapush(rc: ReleaseCohort) -> DependencyGraph:
+    """Convert the Release Cohort's dependency graph (expressed in terms of GitHub URLs)
+    to a MetaPush-style dependency graph (expressed in terms of 04t and 033 package ids)
+    """
+    package_versions_to_github_urls = {}
+    github_urls_to_package_versions = {}
+    dep_graph = DependencyGraph.parse_obj([])
+
+    for release in rc.releases.all():
+        if not release.repo.metapush_enabled:
+            continue
+
+        github_url = release.repo.url
+        # We should have exactly one succeeded Release (Upload Release) build for this Release
+        build = release.build_set.filter(
+            plan__role="release", status=BUILD_STATUSES.success
+        ).first()
+        if not build:
+            raise DependencyGraphError(
+                f"Unable to find Build with role Release on {release}"
+            )
+
+        dep_graph_item = get_package_ids_from_build(build)
+        if not dep_graph_item:
+            raise DependencyGraphError("Unable to source package details from Build")
+        dep_graph.__root__.append(dep_graph_item)
+        package_versions_to_github_urls[dep_graph_item.AllPackageVersionId] = github_url
+        github_urls_to_package_versions[github_url] = dep_graph_item.AllPackageVersionId
+
+    # Now, populate the dependencies in the graph.
+    for dep_graph_item in dep_graph:
+        # Get the GitHub URL for _this_ dependency item
+        github_url = package_versions_to_github_urls[dep_graph_item.AllPackageVersionId]
+        # ... get its dependencies from the Release Cohort's graph...
+        deps = rc.dependency_graph.get(github_url, [])
+        # and convert those GitHub URLs back to created Package Versions
+        for url in deps:
+            package_version = github_urls_to_package_versions.get(url)
+            if not package_version:
+                # This would happen if an intermediate link in the dependency chain
+                # is marked metapush_enabled = False
+                raise DependencyGraphError(
+                    "A dependency package version is missing because one or more repositories are not MetaPush-enabled."
+                )
+            dep_graph_item.Dependencies.append(package_version)
+
+    return dep_graph
+
+
+@job
+def send_to_metapush(rc: ReleaseCohort):
+    _send_to_metapush(rc)
+
+
+def _send_to_metapush(rc: ReleaseCohort):
+    token = settings.METAPUSH_AUTHENTICATION_TOKEN
+    endpoint = settings.METAPUSH_ENDPOINT_URL
+
+    if endpoint and token:
+        # Take our dependency graph (GitHub URL-based) and convert to
+        # MetaPush-style 04t/033 id based. MetaPush does not know
+        # about repositories - just packages.
+
+        try:
+            metapush_graph = convert_dependency_graph_to_metapush(rc)
```

Comment:
Because `convert_dependency_graph_to_metapush()` has a type annotation, the LSP should be able to know the type of `metapush_graph`, so it can provide relevant suggestions and corrections.